Jetpack focus: Enable widgets on jetpack app

ravishanker opened the pull request:
This PR enables home screen widgets on Jetpack app just like it is on WordPress app.  It adds
- Quick Press short cut widget
- Views this week widget

Fixes #17099, #17141, #17142


https://user-images.githubusercontent.com/990349/187333737-a348503a-927d-4f69-9008-369d107f97b8.mov


To test:

Test 1:
- Build and run Jetpack app
- Long press on Jetpack app icon in app launcher or on home screen
- Ensure it show now **Widgets** option in short cuts menu

Test 2:
- Tap on **Widgets** option
- Ensure it pops up a dialog with two widgets; a QuickPress shortcut and Views this week widget

Test 3:
- Tap and hold View this week, then drop it on home screen
- Ensure it pops up  dialog with options for Site and Color selections
- Select a Site from popup
- Select a Color; light or dark from popup
- Tap **ADD WIDGET** button which should be enable after the selections
- Ensure the widget is added to home screen

Test 4:
- You can try resizing the widget
- Widget has a wide view and a block view as shown in the video

Test 5:
- Tap on the widget
- Ensure it launches the Stats screen on insights

Test 6:
- Test Quick Press short cut too, which works just the same as on WordPress app

NOTE:
- The widget set to display **Views**, **Visitors**, **Likes** and **Comments** for this week and matches numbers on Week tab for current week.  We can update this along with other widgets if required in the next PRs.

## Regression Notes
1. Potential unintended areas of impact
Tested to make sure the widgets on WordPress app work as before and not affected by this on Jetpack app

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Tested WordPress app and Jetpack apps separately

3. What automated tests I added (or what prevented me from doing so)
Existing and updated unit tests

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

ravishanker committed:
Enable widgets on home screen

- Enable home screen widgets on Jetpack app
- Add Quick Press Shortcut

ravishanker committed:
Implement widget updater

- Update widget service
- Implement block style widget

ravishanker committed:
Add preview images

- Add and update preview images

ravishanker committed:
Update image assets

ravishanker committed:
Update logic to for this week

ravishanker committed:
Add unit tests

ravishanker committed:
Update RELEASE-NOTES.txt

irfano commented on WordPress/src/jetpack/AndroidManifest.xml:
```diff
@@ -15,6 +16,46 @@
             </intent-filter>
         </activity>
+
+        <activity
+            android:name="org.wordpress.android.ui.AddQuickPressShortcutActivity"
+            android:label="@string/quickpress_shortcut_activity_label"
+            android:theme="@style/WordPress.NoActionBar"
+            android:enabled="true"
+            android:exported="true"
+            tools:replace="android:enabled"
+            tools:ignore="ManifestResource">
+            <intent-filter>
+                <action android:name="android.intent.action.CREATE_SHORTCUT" />
+
+                <category android:name="android.intent.category.DEFAULT" />
+            </intent-filter>
+        </activity>
```
Why did we add AddQuickPressShortcutActivity in jetpack's manifest? It was already in `main/AndroidManifest.xml` but disabled for Jetpack. I remove `android:enabled` line from AddQuickPressShortcutActivity of `main/AndroidManifest.xml` file and it still worked.
Can we just remove `android:enabled` line instead of adding the same activity here?

ravishanker replied:
Good point.  Updated.  However, I thought it would be cleaner when removing it on WP, should make a note of when removing.

irfano reviewed (changes requested):
Good job! I tested widgets and work great.
<img src="https://user-images.githubusercontent.com/2471769/187802315-33a74bb8-d503-4d11-9d3f-44c646219adb.png" height=640> 
- If I add the Views this week widget when I was logged out, "Please log in to WordPress app..." string should be updated to "Please log in to Jetpack app..."
- Will we need translated versions of `stats_weeks_widget_preview.png`?

ravishanker committed:
Update login message

Updated login message for Jetpack app when it is Jetpack

ravishanker committed:
Update for QuickPress shortcut

Removed enabled element in WordPress manifest instead of adding it in Jetpack manifest

ravishanker commented:
>     * "Please log in to WordPress app..." string should be updated to "Please log in to Jetpack app..."
Good catch!  Updated

>     * Will we need translated versions of `stats_weeks_widget_preview.png`?
Probably not and too many that would be.  WP Android doesn't have them either.  A better approach could be separate [preview layouts with fake data for android 12](https://developer.android.com/develop/ui/views/appwidgets/advanced#build-accurate-previews) that we can look at next sprint.

ravishanker committed:
Merge branch 'trunk' into Jetpack-focus-Enable-stats-widgets-on-Jetpack-app

ravishanker committed:
Merge branch 'trunk' into Jetpack-focus-Enable-stats-widgets-on-Jetpack-app

irfano commented:
LGTM! 👍🏻
